akka-actor-tests/src/test/scala/akka/actor/SchedulerSpec.scala

```diff
@@ -483,10 +485,13 @@ class LightArrayRevolverSchedulerSpec extends AkkaSpec(SchedulerSpec.testConfRev
           }
           def delay = if (ThreadLocalRandom.current.nextBoolean) step * 2 else step
           val N = 1000000
-          (1 to N).foreach(_ => sched.scheduleOnce(delay)(counter.incrementAndGet()))
+          (1 to N).foreach(_ =>
+            sched.scheduleOnce(delay, new Scheduler.TaskRunOnClose {
+              override def run(): Unit = counter.incrementAndGet()
+            }))
           sched.close()
           Await.result(terminated, 3.seconds.dilated) should be > 10
-          awaitCond(counter.get == N)
+          awaitAssert(counter.get should ===(N))
```

Comment:
I guess for the original "survive vicious enqueueing" check we could introduce a check that schedules 1000000 'regular' tasks and checks they all trigger within some deadline - but OK